src/common/path_util.cpp

```diff
@@ -82,11 +82,22 @@ static std::filesystem::path GetBundleParentDirectory() {
 
 static auto UserPaths = [] {
 #ifdef __APPLE__
-    std::filesystem::current_path(GetBundleParentDirectory());
+    // Start by assuming the base directory is the bundle's parent directory.
+    std::filesystem::path base_dir = GetBundleParentDirectory();
+    std::filesystem::path user_dir = base_dir / "user";
+
+    // Check if the "user" directory exists in the current path:
+    if (!std::filesystem::exists(user_dir)) {
+        // If it doesn't exist, use the hardcoded path:
+        base_dir = std::filesystem::path(getenv("HOME")) / "Library/Application Support/shadPS4";
```

Comment:
Should probably use the `/` operator of filesystem path instead of hardcoding in the string, e.g. `std::filesystem::path(getenv("HOME")) / "Library" / "Application Support" / "shadPS4"`